Lib/test/support/os_helper.py

```diff
@@ -278,6 +278,48 @@ def skip_unless_working_chmod(test):
     return test if ok else unittest.skip(msg)(test)
 
 
+# Check whether the current effective user has the capability to override
+# DAC (discretionary access control). Typically user root is able to
+# bypass file read, write, and execute permission checks. The capability
+# is independent of the effective user. See capabilities(7).
+_can_dac_override = None
+
+def can_dac_override():
+    global _can_dac_override
+
+    if not can_chmod():
+        _can_dac_override = False
+    if _can_dac_override is not None:
+        return _can_dac_override
+
+    try:
+        with open(TESTFN, "wb") as f:
+            os.chmod(TESTFN, 0o400)
+            try:
+                with open(TESTFN, "wb"):
+                    pass
+            except OSError:
+                _can_dac_override = False
+            else:
+                _can_dac_override = True
+    finally:
+        unlink(TESTFN)
+
+    return _can_dac_override
+
+
+def skip_if_dac_override(test):
+    ok = not can_dac_override()
+    msg = "incompatible with CAP_DAC_OVERRIDE"
+    return test if ok else unittest.skip(msg)(test)
+
+
+def skip_unless_dac_override(test):
+    ok = can_dac_override()
+    msg = "requires CAP_DAC_OVERRIDE"
+    return test if ok else unittest.skip(msg)(test)
```

Comment:
Can we memoize the capacity check? Something like this:

```python
_dac_is_overridable = can_dac_override()


def skip_if_dac_override(test):
    msg = 'incompatible with CAP_DAC_OVERRIDE'
    return test if not _dac_is_overridable else unittest.skip(msg)(test)


def skip_unless_dac_override(test):
    msg = 'requires CAP_DAC_OVERRIDE'
    return test if _dac_is_overridable else unittest.skip(msg)(test)
```

Reply:
The state of DAC override check is already memoized.